show/bgp_common.py

```diff
@@ -0,0 +1,420 @@
+                print "Routing entry for {}".format(str(route))
+                str_2_print = '  Known via "{}", distance {}, metric {}'.format(info[i]['protocol'], info[i]['distance'], info[i]['metric'])
+                if "selected" in info[i]:
+                    str_2_print += ", best"
+                print str_2_print
+                print "  Last update {} ago".format(info[i]['uptime'])
+                for j in range(0, len(info[i]['nexthops'])):
+                    if "directlyConnected" in info[i]['nexthops'][j]:
+                        print "  * directly connected, {}\n".format(info[i]['nexthops'][j]['interfaceName'])
+                    else:
+                        if "ip" in info[i]['nexthops'][j]:
+                            str_2_print = "  * {}".format(info[i]['nexthops'][j]['ip'])
+
+                        if "active" in info[i]['nexthops'][j]:
+                            str_2_print += ", via {}".format(info[i]['nexthops'][j]['interfaceName'])
+                        else:
+                            str_2_print += " inactive"
+                        print str_2_print
+                print
+            else:
+                str_2_print = ""
+                str_2_print += proto_code[info[i]['protocol']]
+                if "instance" in info[i]:
+                    str_2_print += "[" + str(info[i]['instance']) + "]"
+                if "selected" in info[i]:
+                    str_2_print += ">"
+                else:
+                    str_2_print += " "
+                for j in range(0, len(info[i]['nexthops'])):
+                    if j != 0:
+                        str_2_print = "  "
+                    str_2_print += get_status_output_char(info[i], j)
+                    # on 1st nexhop print the prefix and distance/metric if appropriate.
+                    # on all subsequent nexthop replace the prefix and distance/metric by empty spaces only.
+                    if j == 0:
+                        str_2_print += info[i]['prefix'] + get_distance_metric_str(info[i])
+                        str_length = len(str_2_print)
+                    else:
+                        # For all subsequent nexthops skip the spacing to not repeat the prefix section
+                        str_2_print += " "*(str_length - 3)
+                    # Get the nexhop info portion of the string
+                    str_2_print += get_nexthop_info_str(info[i]['nexthops'][j])
+                    # add uptime at the end of the string
+                    str_2_print += " {}".format(info[i]['uptime'])
+                    # print out this string
+                    print str_2_print
+
+
+def merge_to_combined_route(combined_route, route, r_info_l):
+    # The following protocols do not have multi-nexthops. mbrship test with small list is faster than small set
+    # If other protocol are also single nexthop not in this list just add them
+    single_nh_proto = ["connected"]
+    new_info_l = copy.deepcopy(r_info_l)
```

Comment:
why do need to a deepcopy here ?

Reply:
@arlakshm Agreed.  there is no need to have deepcopy here.  will remove it.